roles/config-rh-sso/tasks/setup-rh-sso.yml

```diff
@@ -3,11 +3,67 @@
   import_role:
     name: ../roles/config-packages
 
-- name: 'Update SSO bind addresses'
-  replace:
-    path: /etc/opt/rh/rh-sso7/keycloak/standalone/standalone.xml
-    regexp: '127.0.0.1'
-    replace: '0.0.0.0'
+- name: 'Start Red Hat SSO in Standalone mode'
+  service:
+    name: rh-sso7
+    state: restarted
+
+- name: 'Waiting for SSO service to load'
+  pause:
+    seconds: '30'
+
+- name: 'Update Red Hat SSO management interface IP address'
+  shell: >
+    "/opt/rh/rh-sso7/root/usr/share/keycloak/bin/jboss-cli.sh" -c
+    --command='/interface=management:write-attribute(name=inet-address,value="${jboss.bind.address.management:0.0.0.0}")'
+  args:
+    executable: "/bin/bash"
+
+- name: 'Update Red hat SSO public interface IP address'
+  shell: >
+    "/opt/rh/rh-sso7/root/usr/share/keycloak/bin/jboss-cli.sh" -c
+    --command='/interface=public:write-attribute(name=inet-address,value="${jboss.bind.address:0.0.0.0}")'
+  args:
+    executable: "/bin/bash"
+
+- name: 'Create a keystore to store SSL certificates'
+  java_keystore:
+    name: "{{ sso_keystore_name }}"
+    certificate: "{{lookup('file', '../path/to/file.cer') }}"
+    private_key: "{{lookup('file', '../path/to/file.key') }}"
+    password: "{{ sso_keystore_pass }}"
+    dest: "{{ sso_keystore_path }}"
+
+- name: 'Add a new SSL based security-realm'
+  shell: >
+    "/opt/rh/rh-sso7/root/usr/share/keycloak/bin/jboss-cli.sh" -c
+    --command='/core-service=management/security-realm=UndertowRealm:add()'
+  args:
+    executable: "/bin/bash"
+  register: process_undertowrealm
+  failed_when:
+    - process_undertowrealm.rc == 1
+    - "'WFLYCTL0212' not in process_undertowrealm.stdout"
+
+- name: 'Configure security-realm to use the keystore'
+  shell: >
+    "/opt/rh/rh-sso7/root/usr/share/keycloak/bin/jboss-cli.sh" -c
+    --command='/core-service=management/security-realm=UndertowRealm/server-identity=ssl:add(keystore-path="{{ sso_keystore_path }}", keystore-password="{{ sso_keystore_pass }}")'
+  args:
+    executable: "/bin/bash"
+  failed_when:
+    - process_undertowrealm.rc == 1
```

Comment:
This (and the next) `failed_when` seems to be checking the same outcome/output as the one above. Should probably be checking the command for each of these, right?

Reply:
Yes, these jboss commands will ensure that our https-enabled Realm is in the desired state, however, if we re-run the automation, it will fail with the error code `WFLYCTL0212`, meaning that the Realm is already on the define state. These `failed_when` conditionals will allow the automation to continue to run whether the Realm is already in the desired state.

Reply:
Fixed by adding the line 43 to the two upcoming tasks.